noxfile.py

```diff
@@ -0,0 +1,86 @@
+import nox
+
+nox.options.reuse_existing_virtualenvs = True
+nox.options.default_venv_backend = "uv|virtualenv"
+
+
+@nox.session
+@nox.session(name="tests-cryptography-main")
+@nox.session(name="tests-cryptography-minimum")
+@nox.session(name="tests-wheel")
+@nox.session(name="tests-cryptography-minimum-wheel")
+@nox.session(name="tests-random-order")
+def tests(session: nox.Session) -> None:
+    cryptography_version = None
+    use_wheel = False
+    random_order = False
+
+    if "cryptography-main" in session.name:
+        cryptography_version = "main"
+    elif "cryptography-minimum" in session.name:
+        cryptography_version = "minimum"
+
+    if "wheel" in session.name:
+        use_wheel = True
+
+    if "random-order" in session.name:
+        random_order = True
+
+    session.env.update(
+        {
+            "PIP_NO_BINARY": "" if use_wheel else "cryptography",
+        }
+    )
+
+    deps = ["coverage>=4.2"]
+
+    if cryptography_version == "minimum":
+        deps.append("cryptography==41.0.5")
```

Comment:
Could we set `41.0.5` as a constant at the top of this script to make it easier for future updates? Now that this is just python we can organize it better than the tox ini 😄